src/components/layouts/EvaluationLayout.tsx

```diff
@@ -97,34 +97,13 @@ function EvaluationLayout({
 }: Props): React.ReactElement {
   const intl = useIntl()
 
-  const [existsFeedback, setExistsFeedback] = useState(feedback.length > 0)
-  const [existsConfusion, setExistsConfusion] = useState(confusionTS.length > 0)
-  const [numberOfTabs, setNumberOfTabs] = useState(
-    instanceSummary.length + (existsFeedback ? 1 : 0) + (existsConfusion ? 1 : 0)
-  )
   const [currentIndex, setCurrentIndex] = useState(activeInstance)
-  const [feedbackIndex, setFeedbackIndex] = useState(existsConfusion ? numberOfTabs - 2 : numberOfTabs - 1)
-  const [confusionIndex, setConfusionIndex] = useState(numberOfTabs - 1)
-
-  useEffect(() => {
-    setExistsFeedback(feedback.length > 0)
-  }, [feedback])
 
-  useEffect(() => {
-    setExistsConfusion(confusionTS.length > 0)
-  }, [confusionTS])
-
-  useEffect(() => {
-    setNumberOfTabs(instanceSummary.length + (existsFeedback ? 1 : 0) + (existsConfusion ? 1 : 0))
-  }, [existsFeedback, existsConfusion, instanceSummary])
-
-  useEffect(() => {
-    setFeedbackIndex(existsConfusion ? numberOfTabs - 2 : numberOfTabs - 1)
-  }, [existsConfusion, numberOfTabs])
-
-  useEffect(() => {
-    setConfusionIndex(numberOfTabs - 1)
-  }, [numberOfTabs])
+  const existsFeedback = feedbacks.length > 0
```

Comment:
All of these are derived props, but as the confusion and feedback are only shown if the evaluation has been completed, we will never need to update them. So we can "dehookify" this.